force-app/lazyIterator/classes/IsFieldChangedFrom.cls

```diff
@@ -0,0 +1,24 @@
+/**
+ * @author danb@nebulaconsulting.co.uk
+ * @date 08/04/2022
+ */
+
+global class IsFieldChangedFrom extends TriggerContextBooleanFunction {
+
+    private SObjectField field;
+    private Function function;
+
+    global IsFieldChangedFrom(SObjectField field, Object valueOrFunction) {
+        this.field = field;
+        if(valueOrFunction instanceof Function) {
+            this.function = (Function)valueOrFunction;
+        } else {
+            this.function = new ConstantFunction(valueOrFunction);
+        }
+    }
+
+    global override Boolean isTrueFor(SObject oldRecord, SObject newRecord) {
+        return oldRecord != null && oldRecord.get(field) != newRecord.get(field) && oldRecord.get(field) == function.call(oldRecord);
```

Comment:
IsChangedFrom should return false if the old record doesn't exist and the value we are checking is not null.

For example, if we are checking to see if we've changed from 'Acme' 

`.filterT(new nebc.IsFieldChangedFrom(Account.Name, 'Acme'))`

This would always be false, if there was no old record. But

`.filterT(new nebc.IsFieldChangedFrom(Account.Name, null))`

would be true if the account name was populated in the new record and there was no old record.

I take your point on using the safe navigator though, it will tidy up the code.

I've added an additional test to cover the null value scenario in BooleanFunctionsTest.

Commit is [here](https://github.com/aidan-harding/nebula-core/pull/10/commits/60f0e9c9752d5f370c480061382fe1baa5c19a54).